### PR TITLE
Update to OmniPay 3.0

### PR DESCRIPTION
**Pull request by StewPoll**
This PR updates the library to allow it to be used in Omnipay 3.0

**Code Review:**
- [x] Unit tests exist
- [x] Code formatting standards reviewed (PSR-2, DocBlocks, whitespace)

### CODE REVIEW

File: src/Message/AbstractRequest.php [hunk 1]
```diff
@@ -56,29 +56,34 @@ public function setGivingType($value)
...
+        'POST',
             $url,
-            null,
-            $data,
-            array(
-                'auth' => array($this->getPublicKey(), $this->getPrivateKey())
+            [],
+            json_encode(
+                array_merge(
+                    ['auth' => [$this->getPublicKey(), $this->getPrivateKey()]],
```
Comment (joshmcrae) [inline src/Message/AbstractRequest.php]: The API documentation doesn't indicate that the authentication details can go into the request payload.

Could you please explain the reason for this change?

https://docs.tithe.ly/reference#charge-object

File: src/Message/AbstractRequest.php [hunk 2]
```diff
@@ -56,29 +56,34 @@ public function setGivingType($value)
...
-                'auth' => array($this->getPublicKey(), $this->getPrivateKey())
-            )
-        )->send();
+        $url = $this->getEndpoint() . '/organizations/' . $this->getOrganizationId();
+        $httpResponse = $this->httpClient->request(
+            'GET',
+            $url,
+            [],
+            json_encode([
+                'auth' => [$this->getPublicKey(), $this->getPrivateKey()]
```
Comment (joshmcrae) [inline src/Message/AbstractRequest.php]: As above.

**Review by joshmcrae — CHANGES_REQUESTED** [review]
Tests are failing on Travis due to PHP version. Since 5.6 is the lowest supported, versions below this should be removed from the build.

**Comment by StewPoll** [comment]
@joshmcrae 
After looking at the docs, and how a few other similar plugins are doing it, I can see no reason for this.

I've made some changes on behalf of Ryan.

**Review by joshmcrae — CHANGES_REQUESTED** [review]
There are still failing tests. I've updated the Travis CI configuration to include only the relevant PHP versions.

There appears to be a compatibility issue with PHPUnit.

**Review by joshmcrae — APPROVED** [review]

**Pull request merged**